.gitignore

```diff
@@ -138,3 +138,4 @@ Desktop.ini
 *~
 .directory
 .merge_file*
+.idea
```

Comment:
The idea of committing a subset of `.idea` was to streamline inspections and formatting, mostly. I generally tend to do this everywhere, so a dev can jump in and have the environment ready with no additional processes needed.

But if you feel like it doesn't suit the repo, feel free to remove those files.

Reply:
I tend to have all the .idea configuration generated by gradle. I just thought these files were committed by mistake

Reply:
> I tend to have all the .idea configuration generated by Gradle

A part of those files are indeed fine as generated at project import time, but others like `codeStyles/*` or `inspectionProfiles/*` are not generated, and when opening the project you'll find the "Default" settings applied. This means everyone will add his own custom formatting.

Reply:
I would avoid KLint and stick with `editorconf` then. I can generate the `.editorconfig` file in case.  
KLint configuration is a PITA and it doesn't have a reliable IDE plugin.

Reply:
Great, let's go with .editorconfig then!
Should I consider this a follow up task and proceed with this merge or should we add this to this PR?

Reply:
Not sure if I can add it, but you can generate it:

1. Restore the files you deleted
2. Open up the IntelliJ Settings (checking it says "Project" as scheme) and export as EditorConfig

   ![image](https://github.com/Strumenta/antlr-kotlin/assets/19871649/490d2dd6-c722-4a17-ada0-c588c3f3d79b)

3. Select the root of the repository as export target
4. Delete the `.idea` files again